src/components/rule-builder/rule-builder.component.tsx

```diff
@@ -0,0 +1,1746 @@
+        case 'BSA':
+          return `calcBSA(${argumentsSchema})`;
+        case 'Height For Age Zscore':
+          return `calcHeightForAgeZscore(${argumentsSchema})`;
+        case 'BMI For Age Zscore':
+          return `calcBMIForAgeZscore(${argumentsSchema})`;
+        case 'Weight For Height Zscore':
+          return `calcWeightForHeightZscore(${argumentsSchema})`;
+        case 'Expected Delivery Date':
+          return `calcEDD(${argumentsSchema})`;
+        case 'Months On ART':
+          return `calcMonthsOnART(${argumentsSchema})`;
+        case 'Age Based On Date':
+          return `calcAgeBasedOnDate(${argumentsSchema})`;
+        case 'Time Difference in days':
+          return `calcTimeDifference(${argumentsSchema}, 'd')`;
+        case 'Time Difference in weeks':
+          return `calcTimeDifference(${argumentsSchema}, 'w')`;
+        case 'Time Difference in months':
+          return `calcTimeDifference(${argumentsSchema}, 'm')`;
+        case 'Time Difference in years':
+          return `calcTimeDifference(${argumentsSchema}, 'y')`;
+        case 'Viral Load Status':
+          return `calcViralLoadStatus(${argumentsInExpression.join(', ')})`;
+        case 'Next Visit Date':
+          return `calcNextVisitDate(${argumentsSchema})`;
+        case 'Treatment End Date':
+          return `calcTreatmentEndDate(${argumentsSchema})`;
+        case 'Gravida':
+          return `calcGravida(${argumentsSchema})`;
+      }
+    };
+
+    /**
+     * Build the schema for a "historical" expression.
+     * Constructs the appropriate expression for a historical condition based on the provided parameters.
+     */
+    const getHistoricalExpressionSchema = useCallback(
+      (
+        condition: string,
+        actionConcept: string,
+        selectedAnswers?: Array<{
+          concept: string;
+          label: string;
+        }>,
+        targetConcept?: string,
+      ) => {
+        const selectedConcepts = selectedAnswers?.map((item) => `'${item.concept}'`).join(', ');
+        switch (condition) {
+          case 'Is Empty':
+            return `_.isEmpty(HD.getObject('prevEnc').getValue('${actionConcept}')) ? undefined : HD.getObject('prevEnc').getValue('${actionConcept}')`;
+          case 'Not Empty':
+            return `!_.isEmpty(HD.getObject('prevEnc').getValue('${actionConcept}')) ? undefined : HD.getObject('prevEnc').getValue('${actionConcept}')`;
+          case 'Contains any':
+            return `arrayContainsAny([${selectedConcepts}], HD.getObject('prevEnc').getValue('${targetConcept}')) ? '${actionConcept}' : HD.getObject('prevEnc').getValue('${actionConcept}')`;
+        }
+      },
+      [],
+    );
+
+    /**
+     * Retrieve the arguments required for calculation functions.
+     * Identifies and returns the necessary parameters based on the type of calculation.
+     */
+    const getArguments = useCallback((expression: string) => {
+      switch (expression) {
+        case 'BMI':
```

Comment:
In general, we seem to have a lot of these that switch on labels rather than some sort of language-independent key, which means that this makes it harder to (eventually) translate this part of the app.